src/main/java/wolox/training/models/Book.java

```diff
@@ -16,140 +15,97 @@
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
 import javax.persistence.ManyToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
 
 @Entity
+@NoArgsConstructor
+@Getter
 public class Book {
```

Comment:
For all the "non empty" validations made on the setters, you could add the annotation `@NotBlank` to the corresponding attributes. If I remember well, the `@NotBlank` also catches de "null" values.